source/index.html.md

```diff
@@ -287,16 +514,103 @@ When you change to a new video (in the same player) you need to update the infor
+  error_message: 'Description of error'
+});
+```
+
+```jwplayer
+player.mux.emit('error', {
+  error_code: 100,
+  error_message: 'Description of error'
+});
+```
+
+```bitmovin
+player.mux.emit('error', {
+  error_code: 100,
+  error_message: 'Description of error'
+});
+```
+
+```ooyala
+player.mux.emit('error', {
+  error_code: 100,
+  error_message: 'Description of error'
+});
+```
+
+<p class="lang-specific video-element videojs jwplayer bitmovin ooyala">
+Mux currently tracks fatal errors that are triggered by the player in use. However, in some cases, such as third-party HLS libraries for web players, fatal errors may occur outside of the context of the player. To handle these cases, Mux exposes a type of event that can be triggered via javascript. <strong>This method should only be used for fatal errors. Do not use this for "errors" that do result in playback failure</strong>
+<br><br>
+When triggering an error event, it is important to provide an <code>error_code</code> and <code>error_message</code> in an object as the second parameter. The <code>error_message</code> should provide a detailed description of the error as it happened, whereas the <code>error_code</code> must be an integer, and should provide a category of the error. If the errors match up with the <a href="https://developer.mozilla.org/en-US/docs/Web/API/MediaError">HTML Media Element Error</a>, you can use the same codes as the corresponding HTML errors. However, for custom errors, you should choose a number greater than or equal to 100. Whether all errors use the same code or you use different codes for different types of errors is up to you, but in general you should not send a distinct code for each possible error message.
+</p>
+
+<p class="lang-specific objective-c">
+There is currently no way to fire custom error events for iOS/tvOS. Mux automatically tracks any errors that are triggered by the underlying video player.
+</p>
+
+## Ad Support
+
+<p class="lang-specific video-element">
+There are currently no ad integrations supported for the core HTML Video Element integration. Ads will work fine, but will not be tracked appropriately.
+</p>
+
+<p class="lang-specific videojs">
+Mux supports <a href="https://github.com/googleads/videojs-ima">videojs-ima</a>, as well as Brightcove's IMA3, FreeWheel, and OnceUX plugins. Simply configure these plugins as you would normally, and Mux will track ads automatically. No additional configuration is needed.
+</p>
+
+<p class="lang-specific jwplayer">
+Mux has been tested with and supports JW Player's <code>vast</code> integration for pre-, mid-, and post-roll ads. Simply configure these plugins as you would normally, and Mux will track ads automatically. No additional configuration is needed.
+<br><br>
+Other JW Player ad integrations, such as <code>googima</code> and <code>freewheel</code> have not been tested, but may work out of the box. Please <a href="mailto:help@mux.com">contact us</a> with any questions.
```

Comment:
well, I did that on purpose actually. The sales-contact links are for people who may have stumbled on our docs and want new players we don't support now (they're new leads). The others were more for current customers that actually ran into issues where the ad libraries didn't work correctly (hence support).

Thoughts?